mos_tests/ironic/conftest.py

```diff
@@ -93,3 +95,92 @@ def baremetal_node(env_name, suffix):
     node.erase()
     disk.volume.erase()
     disk.delete()
+
+
+@pytest.yield_fixture
+def keypair(os_conn):
+    keypair = os_conn.create_key(key_name='ironic-key')
+    yield keypair
+    os_conn.delete_key(key_name=keypair.name)
+
+
+@pytest.yield_fixture
+def flavor(baremetal_node, os_conn):
+    flavor = os_conn.nova.flavors.create(name=baremetal_node.name,
+                                         ram=baremetal_node.memory,
+                                         vcpus=baremetal_node.vcpu,
+                                         disk=settings.IRONIC_DISK_GB)
+
+    yield flavor
+    flavor.delete()
+
+
+@pytest.yield_fixture(scope='module')
+def image_file():
+    image_file = common.gen_temp_file(prefix='image', suffix='img')
+    yield image_file
+    image_file.unlink(image_file.name)
+
+
+@pytest.yield_fixture
+def ubuntu_image(os_conn, image_file):
```

Comment:
This fixture registers, returns, and delete after test Ubuntu image, so i believe that `ubunut_image` is more suitable than  os_image or main_image. About other images - we can make it with different fixtures (like windows_image or centos_image), or with parametrization (in this case - fixture will be renamed).